src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,125 @@
+            var cancellationToken = context.CancellationToken;
+            var workspace = document.Project.Solution.Workspace;
+            var semanticModel = await document.GetSemanticModelForSpanAsync(new TextSpan(position, length: 0), cancellationToken).ConfigureAwait(false);
+
+            var token = TryGetOpenBraceOrCommaInPropertyPattern(document, semanticModel, position, cancellationToken);
+            if (token == default)
+            {
+                return;
+            }
+
+            var pattern = (PatternSyntax)token.Parent.Parent;
+            var type = semanticModel.GetTypeInfo(pattern).ConvertedType;
+            if (type == null)
+            {
+                return;
+            }
+
+            // Find the members that can be tested.
+            IEnumerable<ISymbol> members = semanticModel.LookupSymbols(position, type);
+            members = members.Where(m => m.CanBeReferencedByName &&
+                IsFieldOrReadableProperty(m) &&
+                !m.IsImplicitlyDeclared);
+
+            // Filter out those members that have already been typed
+            var alreadyTestedMembers = GetTestedMembers(semanticModel.SyntaxTree, position, cancellationToken);
+            var untestedMembers = members.Where(m => !alreadyTestedMembers.Contains(m.Name) &&
+                m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
+
+            foreach (var untestedMember in untestedMembers)
+            {
+                context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
+                    displayText: untestedMember.Name.EscapeIdentifier(),
+                    insertionText: null,
+                    symbols: ImmutableArray.Create(untestedMember),
+                    contextPosition: token.GetLocation().SourceSpan.Start,
+                    rules: s_rules));
+            }
+        }
+
+        private static bool IsFieldOrReadableProperty(ISymbol symbol)
+        {
+            if (symbol.IsKind(SymbolKind.Field))
+            {
+                return true;
+            }
+
+            if (symbol.IsKind(SymbolKind.Property) && !((IPropertySymbol)symbol).IsWriteOnly)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static readonly CompletionItemRules s_rules = CompletionItemRules.Create(enterKeyRule: EnterKeyRule.Never);
+
+        internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
+            => CompletionUtilities.IsTriggerCharacter(text, characterPosition, options) || text[characterPosition] == ' ';
+
+        private static SyntaxToken TryGetOpenBraceOrCommaInPropertyPattern(
```

Comment:
nit: should be named TryGetOpenBraceOrCommaInProperty**Sub**pattern because it doesn't actually have to be inside a PropertyPattern